### PR TITLE
move to verification

### LINKED ISSUE #1334: switch statistical tests of mutation to verification.py
The tests that our mutation model are working are statistical, perhaps unavoidably, [and brittle](https://github.com/tskit-dev/msprime/pull/1314/files#r534381142). This is a consquence of using p-values - multiple testing and so forth. And there's no good way around this, really, since: suppose we do N tests, then setting the p-value threshold at 0.01/N should result in a fail only 1% of the time... but, suppose we change a bit of code that only affects 5 of those tests. Now our p-value threshold is far too small, since really we'd want to set it at 0.01/5 (say). Also, even a 1% failure is going to get triggered too often.

But! A much better test would be to ask if the statistics say the deviation from expected is no more than something small. That'd mean replacing e.g. [this bit](https://github.com/tskit-dev/msprime/blob/5ca4eb490d12203e0ec791f655ddeda2fae3e254/tests/test_mutations.py#L986):
```
            chisq = stats.chisquare(obs[not_zeros], exp[not_zeros])
            assert chisq.pvalue > p_th
```
with something like
```
            assert np.max(np.abs(obs[not_zeros] - exp[not_zeros])/exp[not_zeros] < 0.01)
```
to say that none of the categories are off by more than 1%. If a test fails, we would just need to increase the sample size (e.g., do a longer sequence).

### PR DESCRIPTION
**Pull request by petrelharp**
Closes #1334. I think I've got all of @GertjanBisschop's great tests moved over.

Some of the plots are (unavoidably) not qq plots, but plots of relative error as a function of sample size, that we expect to be going down towards zero, like this one:
F84_continuous_transitions

How's this look?

### CODE REVIEW

File: verification.py [hunk 1]
```diff
@@ -4087,6 +4087,285 @@ def test_simulate_from_demography(self):
...
+    p_threshold = 0.001
+
+    def chisquare(self, observed, expected, p_th=p_threshold):
+        obs = observed.reshape(np.product(observed.shape))
+        exp = expected.reshape(np.product(observed.shape))
+        assert np.all(obs[exp == 0] == 0)
+        not_zeros = exp > 0
+        if sum(not_zeros) > 1:
+            chisq = scipy.stats.chisquare(obs[not_zeros], exp[not_zeros])
+            assert chisq.pvalue > p_th
```
Comment (jeromekelleher) [inline verification.py:4101]: What happens if this assertion fails? Does it bring everything down with a clear error message when running with multiple threads? I just want to be sure that I don't miss a problem if this ever fails.

**Review by jeromekelleher — COMMENTED** [review]
Looks great, thanks @petrelharp. Would you mind putting a hline at y=0 in the left plot please (maybe the right one too), just so it's obvious what the values should be tending to? I need to scan hundreds of plots when doing these checks, so it really helps if they are "right" in as obvious a way as possible.

File: verification.py (hunk 1 above)
Comment (petrelharp) [reply verification.py:4101]: Whoops - didn't even mean to leave this in here. I could put these p-value checks back in, but they *should* be redundant with the figures.

Comment (jeromekelleher) [reply verification.py:4101]: It's up to you - might be best to remove them and keep things simple.

**Comment by petrelharp** [comment]
Ok, how's this?
F84_continuous_roots

I added the 1/sqrt(n) line on the right; the visual check here is that the blue line is "going down", and the dotted line is the guide for how much it should be going down. It's not expected to be always below it, or always on top of it or anything. Is this clear enough?

**Review by jeromekelleher — APPROVED** [review]
LGTM! Quick squasharoo and we're ready to merge.

**Comment by petrelharp** [comment]
Whoops, sorry - I was just finishing up some minor fixing - I got rid of some divide-by-zero warnings, and realized one of the comparisons could be made better (the things we were comparing were in pairs, so a y=x plot is better than a qq plot). I'll put that in a new PR, I guess.

**Pull request merged**